me.gladwell.eclipse.m2e.android/src/main/java/me/gladwell/eclipse/m2e/android/configuration/AttachDocumentsClasspathLoader.java

```diff
@@ -29,7 +30,7 @@ public AttachDocumentsClasspathLoader(@PrunePlatformProvidedDependencies Classpa
     }
 
     @Override
-    public Iterable<IClasspathEntry> load(IJavaProject project) throws FileNotFoundException {
+    public Iterable<IClasspathEntry> load(IJavaProject project) throws FileNotFoundException, CoreException {
```

Comment:
You can just wrap any weird, checked exceptions like `CoreException` in the unchecked `ProjectConfigurationException`. This means you don't need to modify a bunch of unrelated classes when you throw it.